crates/matrix-sdk-base/src/sliding_sync.rs

```diff
@@ -402,6 +405,35 @@ impl BaseClient {
             }
         }
     }
+
+    pub(crate) fn deserialize_state_events_from_timeline(
+        raw_events: &[Raw<AnySyncTimelineEvent>],
+    ) -> Vec<(Raw<AnySyncStateEvent>, AnySyncStateEvent)> {
+        raw_events
+            .iter()
+            .filter_map(|raw_event| {
+                if raw_event.get_field::<serde::de::IgnoredAny>("state_key").is_ok() {
+                    match raw_event.deserialize() {
```

Comment:
You can use `deserialize_as::<AnySyncStateEvent>()` here to avoid the unreachable branch.

Reply:
Indeed, thanks!